go/worker/storage/committee/node.go

```diff
@@ -615,8 +588,120 @@ func (n *Node) flushSyncedState(summary *blockSummary) uint64 {
 	return n.syncedState.LastBlock.Round
 }
 
+func (n *Node) updateExternalServicePolicy(rtComputeNodes runtimeCommittee.NodeDescriptorLookup) {
+	// Create new storage gRPC access policy for the current runtime.
+	policy := accessctl.NewPolicy()
+
+	// Add policy for configured sentry nodes.
+	for _, addr := range n.workerCommonCfg.SentryAddresses {
+		sentryNodesPolicy.AddPublicKeyPolicy(&policy, addr.PubKey)
+	}
+
+	executorCommitteePolicy.AddRulesForNodeRoles(&policy, rtComputeNodes.GetNodes(), node.RoleComputeWorker)
+
+	// TODO: Query registry only for storage nodes after
+	// https://github.com/oasisprotocol/oasis-core/issues/1923 is implemented.
+	nodes, err := n.commonNode.Consensus.Registry().GetNodes(n.ctx, consensus.HeightLatest)
+	if err != nil {
+		n.logger.Error("couldn't get nodes from registry", "err", err)
+	}
+	if len(nodes) > 0 {
+		// Only include storage nodes for our runtime.
+		var storageNodes []*node.Node
+		for _, nd := range nodes {
+			if nd.GetRuntime(n.commonNode.Runtime.ID()) != nil && nd.HasRoles(node.RoleStorageWorker) {
+				storageNodes = append(storageNodes, nd)
+			}
+		}
+		storageNodesPolicy.AddRulesForNodeRoles(&policy, storageNodes, node.RoleStorageWorker)
+	}
+
+	// Update storage gRPC access policy for the current runtime.
+	n.grpcPolicy.SetAccessPolicy(policy, n.commonNode.Runtime.ID())
+	n.logger.Debug("set new storage gRPC access policy", "policy", policy)
+}
+
+func (n *Node) runtimeNodesWatcher() {
+	defer close(n.rtWatcherQuitCh)
+
+	// Watch registry for runtime node updates and update external gRPC policies.
+	// Policy updates are made on:
+	// * any updates to the runtime executor committee nodes
+	// * any updates to the registered storage nodes for the runtime
+	//   (this includes nodes not in committee)
+
+	n.logger.Info("starting runtime nodes watcher")
+
+	rtID := n.commonNode.Runtime.ID()
+	committeeWatcher, err := runtimeCommittee.NewWatcher(
+		n.ctx,
+		n.commonNode.Consensus.Scheduler(),
+		n.commonNode.Consensus.Registry(),
+		rtID,
+		scheduler.KindComputeExecutor,
+		runtimeCommittee.WithAutomaticEpochTransitions(),
+	)
+	if err != nil {
+		n.logger.Error("worker/storage: failed to establish executor committee watcher",
+			"err", err,
+		)
+		return
+	}
+	committeeCh, committeeSub, err := committeeWatcher.Nodes().WatchNodeUpdates()
```

Comment:
Could we use `n.Group.Nodes().WatchNodeUpdates()` instead? This would avoid spawning another committee watcher.

If you only want executor nodes, you could also filter it by doing something like:
```golang
committee.NewFilteredNodeLookup(nodes, committee.TagFilter(tagForCommittee(scheduler.KindExecutor)))
```
(would need to expose `tagForCommittee` though)

Reply:
good idea, done :+1: